interpreter/per-host/src/main/scala/io/buoyant/transformer/perHost/LocalhostTransformer.scala

```diff
@@ -0,0 +1,43 @@
+package io.buoyant.transformer.perHost
+
+import com.twitter.finagle.Name.Bound
+import com.twitter.finagle.{Addr, Address, Name, NameTree}
+import com.twitter.util.Activity
+import io.buoyant.namer.{DelegateTree, DelegatingNameTreeTransformer}
+import java.net.InetAddress
+
+/**
+  * The localhost transformer filters the list of addresses down to only
+  * addresses that have the same IP address as localhost.  This can be used by
+  * an incoming router to only route traffic to local destinations.
+  */
+class LocalhostTransformer extends DelegatingNameTreeTransformer {
+
+  private[this] val localhost = InetAddress.getLocalHost.getAddress
```

Comment:
Our ability to handle those edge cases depends on InetAddress' ability to handle those edge cases.  In some of those situations, this transformation may not make sense or be applicable.  

Yes, I will be adding a k8s-local transformer in a future review that does the same thing but changes out the `isLocal` logic for a /24 subnet comparison.  No need to prematurely refactor.